doc/sphinx/how-to/how-to-test-mir-for-a-release.md

```diff
@@ -0,0 +1,153 @@
+do not fail to consider how your additions may be automated in the future, as
+this is the end goal of our release testing.
+
+Note that all of these tests must be run _after_ a release branch is created
+in the Mir repository, such that the repository is updated beforehand.
+
+## Setup
+We will run the test plan on the `miral-app` test compositor.
+
+1. Install mir from the release candidate PPA
+    ```sh
+    sudo add-apt-repository --update ppa:mir-team/rc
+    sudo apt install mir-demos mir-platform-graphics-virtual mir-graphics-drivers-desktop mir-test-tools
+    ```
+2. Confirm that you can run `miral-app`
+
+## Tests
+Each test must be performed across a combination of different display
+platforms and Ubuntu releases. The following matrix provides the environments
+in which we need to test:
+
+|                                | 24.04    | 24.10      |
+|--------------------------------|----------|------------|
+| gbm-kms                        |          |            |
+| eglstream-kms                  |          |            |
+| eglstream-kms + gbm-kms hybrid |          |            |
+| x11                            |          |            |
+| wayland                        |          |            |
+| virtual                        |          |            |
+
+
+To check which display platform we've selected, we can run `miral-app`
+and grep for the platform string as follows:
+
+```sh
+miral-app | grep "Selected display driver:"
+```
+
+Given the types of outputs that you have configured in your environment,
+you should encounter one of the following scenarios for each output:
+
+1. When NOT on an Nvidia platform and NOT in a hosted environment,
+   then `mir:gbm-kms` is selected
+2. When you have an Nvidia card connected to an output _and_ the system
+   is using Nvidia's proprietary drivers, then `mir:eglstream-kms`
+   is selected
+3. When you are running the compositor hosted in a session that supports X11,
+   then `mir:x11` is selected
+4. When you are running the compositor hosted in a session that supports wayland
+   _and_ you force Mir to use the `mir:wayland` platform using:
+   ```
+   miral-app --wayland-host=$WAYLAND_DISPLAY
+   ```
+   then `mir:wayland` is selected.
+
+5. Check that the virtual platform can run and you can connect to it via a VNC:
+   ```sh
+   WAYLAND_DISPLAY=wayland-1 miral-app --platform-display-lib=mir:virtual \
+       --virtual-output=1280x1024 \
+       --add-wayland-extension=zwp_virtual_keyboard_manager_v1:zwlr_virtual_pointer_manager_v1:zwlr_screencopy_manager_v1
+   ```
+   After, in a separate VT, connect to the VNC:
+   ```sh
+   WAYLAND_DISPLAY=wayland-1 ubuntu-frame-vnc
+   ```
+   Then, check that you can connect via a VNC viewer:
+   ```sh
+   gvncviewer localhost
+   ```
+
+### Run Compositors that are Built for Testing
+1. Run the `mir-test-smoke-runner` and confirm that the final output is:
+   `Smoke testing complete with returncode 0`.
```

Comment:
We should state explicitly how to control the platform selected by the smoke tests.

E.g. testing on the virtual platform doesn't need to connect a VNC client (as described in the previous section), and it is easiest to use environment variables to select the platform:

```bash
MIR_SERVER_PLATFORM_DISPLAY_LIBS=mir:virtual MIR_SERVER_VIRTUAL_OUTPUT=1280x1024 mir-smoke-test-runner
```

Reply:
For consistency, it might be better to use environment variables throughout.

Reply:
I kept a few as commandline params (e.g. `--console-providers` and `--vt`, but the others are now env variables)